blog/webassembly_interaction_patterns.md

```diff
@@ -0,0 +1,102 @@
+---
+slug: webassembly-patterns-command-reactor-library
+title: "WebAssembly Interaction Patterns: Command, Reactor, Library"
+--image: "/img/there-and-back-again.jpg"
+date: 2023-06-06T09:00:00+05:00
+author: "Kevin Hoffman"
+author_profile: "https://mastodon.world/@autodidaddict"
+categories: ["webassembly", "components", "patterns"]
+description: Exploring WebAssembly Interaction Patterns - Command, Reactor, and Library"
+draft: false
+---
+The first thing people need to learn when encountering WebAssembly for the first time is how to interact with the code inside the module.
+There are a number of patterns available for this, and in this blog post I'll cover the main three: _Command, Reactor_, and _Library_.
+
+<!-- truncate -->
+
+As we've mentioned in a number of posts and conference talks, WebAssembly's isolated sandbox forces people to use functions that
+can only accept and return numbers. _Bailey Hayes_ likes to refer to this aspect of WebAssembly as just _"[a bunch of numbers in a trenchcoat](https://youtu.be/6_BRLqxiZPU?t=459)"_. There are 
+plenty of libraries, wrappers, tools, and extensions that try and hide this basic fact, but it's important to understand underneath all of that, it's comes down to exchanging numbers.
+
+Once we get past the fact that we're only dealing with numbers, we can then move on to explore _how_ we're going to interact with the module. Let's take a look at each of these patterns in turn.
+
+## The Command Pattern
+In this pattern, a `command` is a bundle of logically connected information. This command represents an _imperative_: a request that work be performed. Generally
+speaking, the command pattern involves sending a request and then obtaining a response. The defining characteristics of this pattern are the single execution of the command and
+the request/response nature of the interaction.
+
+Commands can be anything from a simple request to add two numbers together to a complex request to perform object recognition on an image frame. The mechanics of getting this 
+command into a WebAssembly module and getting the response back usually involves building the module to target **WASI** and accepting a single command input over **stdin** and delivering
+the response via **stdout**.
+
+The primary appeal of this pattern is simplicity and availability. Even languages that struggle to produce freestanding WebAssembly modules (e.g. targeting `wasm32-unknown-unknown`) can often 
+produce `wasm32-wasi` modules. Accepting data via **stdin** means that the WASI plumbing is taking care of marshaling the data through linear memory and the developer doesn't have to do that manually. Returning the result is a simple matter of writing to **stdout**.
```

Comment:
Adjust to 'takes' care of marshaling; change and to so; change that to it:

Accepting data via stdin means that the WASI plumbing takes care of marshaling the data through the linear memory so the developer doesn't have to do it manually.